From 904a85173fab816bb3c30652300efa93f2333657 Mon Sep 17 00:00:00 2001
From: Alex Bee <knaerzche@gmail.com>
Date: Sun, 15 Dec 2024 01:16:13 +0100
Subject: [PATCH 11/11] HACK: avcodec: Don't check VIDIOC_ENUM_FRAMESIZES
 step_width / step_height

Currently required, since hantro and rkvdec drivers expose wrong values.

Signed-off-by: Alex Bee <knaerzche@gmail.com>
---
 libavcodec/v4l2_request_probe.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/libavcodec/v4l2_request_probe.c b/libavcodec/v4l2_request_probe.c
index 941042ec04..b3b1af3744 100644
--- a/libavcodec/v4l2_request_probe.c
+++ b/libavcodec/v4l2_request_probe.c
@@ -258,9 +258,7 @@ static int v4l2_request_try_framesize(AVCodecContext *avctx,
         } else if ((frmsize.type == V4L2_FRMSIZE_TYPE_STEPWISE ||
                     frmsize.type == V4L2_FRMSIZE_TYPE_CONTINUOUS) &&
                    avctx->coded_width <= frmsize.stepwise.max_width &&
-                   avctx->coded_height <= frmsize.stepwise.max_height &&
-                   avctx->coded_width % frmsize.stepwise.step_width == 0 &&
-                   avctx->coded_height % frmsize.stepwise.step_height == 0) {
+                   avctx->coded_height <= frmsize.stepwise.max_height) {
             return 0;
         }
 
-- 
2.47.2

